### PR TITLE
WebUI: Use Mustache.jl templates

### PR DESCRIPTION
**Pull request by nkottary**

### CODE REVIEW

File: src/webui/webutils.jl [hunk 1]
```diff
@@ -13,43 +13,19 @@ function getcookie(r::HTTP.Request, key::AbstractString, default="")
     return ind === nothing ? default : cookies[ind].value
 end
 
+readtpl(tpl) = Mustache.template_from_file(joinpath(dirname(@__FILE__), "templates", tpl))
```
Comment (christopher-dG) [inline src/webui/webutils.jl]:
```suggestion
readtpl(tpl::AbstractString) = Mustache.template_from_file(joinpath(@__DIR__), "templates", tpl))
```

File: src/webui/webutils.jl [hunk 2]
```diff
@@ -13,43 +13,19 @@ function getcookie(r::HTTP.Request, key::AbstractString, default="")
...
+const INDEX_TPL = readtpl("index.tpl")
+const SELECT_TPL = readtpl("select.tpl")
```
Comment (christopher-dG) [inline src/webui/webutils.jl]: I'm pretty sure making these things `const` means that they get compiled in and won't respond to updates in stuff like `route_prefix` etc.

File: src/webui/webutils.jl [hunk 3]
```diff
@@ -13,43 +13,19 @@ function getcookie(r::HTTP.Request, key::AbstractString, default="")
...
-          <body>
-            <h1><a href="$(ROUTES[:INDEX])">Registrator</a></h1>
-            <h4>Registry URL: <a href="$registry" target="_blank">$registry</a></h3>
-            <h3>Click <a href="$DOCS" target="_blank">here</a> for usage instructions</h3>
-            <br>
-            $body
-          </body>
-        </html>
-        """
+    doc = render(INDEX_TPL,
```
Comment (christopher-dG) [inline src/webui/webutils.jl]: I would suggest:
```jl
render(
    INDEX_TPL;
    route_index=...,
)
```

File: src/webui/webutils.jl (hunk 1 above)
Comment (christopher-dG) [reply src/webui/webutils.jl]: Also, are you sure you don't want `render_from_file`? I'm not positive myself but this is the exported function with a docstring.

File: src/webui/webutils.jl (hunk 2 above)
Comment (nkottary) [reply src/webui/webutils.jl]: We pass routes as args to the templates while rendering so this is ok.

File: src/webui/webutils.jl [hunk 4]
```diff
@@ -13,43 +13,21 @@ function getcookie(r::HTTP.Request, key::AbstractString, default="")
...
-          <body>
-            <h1><a href="$(ROUTES[:INDEX])">Registrator</a></h1>
-            <h4>Registry URL: <a href="$registry" target="_blank">$registry</a></h3>
-            <h3>Click <a href="$DOCS" target="_blank">here</a> for usage instructions</h3>
-            <br>
-            $body
-          </body>
-        </html>
-        """
+    doc = render_from_file(
```
Comment (christopher-dG) [inline src/webui/webutils.jl]: Hah, for future reference this is what I mean every time I bring it up:

```julia
doc = render_from_file(
    INDEX_TPL,  # Four spaces indented
    # ...
)  # Zero spaces indented
```

File: src/webui/routes/select.jl [hunk 5]
```diff
@@ -1,88 +1,10 @@
...
-            <br>
-            """
-    end
-
-    body *= """
-        <button id="submitButton" onclick="do_register()">Submit</button>
-        </div>
-        """
-
+    body = render_from_file(
```
Comment (christopher-dG) [inline src/webui/routes/select.jl]: same thing here -- don't worry about changing it now but just keep it in mind :slightly_smiling_face:

**Pull request merged**